cmd/eksctl-anywhere/cmd/internal/commands/artifacts/download.go

```diff
@@ -0,0 +1,83 @@
+package artifacts
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/aws/eks-anywhere/pkg/logger"
+	"github.com/aws/eks-anywhere/pkg/version"
+	releasev1 "github.com/aws/eks-anywhere/release/api/v1alpha1"
+)
+
+type Reader interface {
+	ReadImages(eksaVersion string) ([]releasev1.Image, error)
+	ReadCharts(eksaVersion string) ([]releasev1.Image, error)
+	ReadBundlesForVersion(version string) (*releasev1.Bundles, error)
+}
+
+type ImageMover interface {
+	Move(ctx context.Context, artifacts ...string) error
+}
+
+type ChartDownloader interface {
+	Download(ctx context.Context, artifacts ...string) error
+}
+
+type Packager interface {
+	Package(folder string, dstFile string) error
+}
+
+type Download struct {
+	Reader           Reader
+	Version          version.Info
+	ImageMover       ImageMover
+	ChartDownloader  ChartDownloader
+	Packager         Packager
+	TmpDowloadFolder string
+	DstFile          string
+}
+
+func (d Download) Run(ctx context.Context) error {
+	if err := os.MkdirAll(d.TmpDowloadFolder, os.ModePerm); err != nil {
```

Comment:
Interesting
So use create a filewriter so the folder gets initialized? Because I'm not writing any file directly (with `[]byte`).

Or do you want to integrate the FileWriter with the `Tar` method?